build_matrix_A.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+
+
+def build_matrix_A(material_property_library, mesh, time):
+    """ This function will take a previously generated mesh and create the 'A'
+    matrix used in solving this problem.
+
+    inputs:
+    --------
+    - material_property_library: this will be a dictionary or nested dictionary
+    containing all of the important data for each material (fuel,
+    various layers of graphite)
+    - mesh: the physical location of each mesh point
+    - materials: the materials in each region and how many nodes are assigned to that material
+    outputs. List of tuples of format (material, number of nodes)
+    - time: An array of timesteps taken during transients. If steady state, supply 0.
+    --------
+    - An array that can be plugged into a subsequent function to solve the
+      A.x=b formula
+    """
+    k = np.zeros(len(mesh))
+    rho = np.zeros(len(mesh))
+    c = np.zeros(len(mesh))
+    DR = np.zeros(len(mesh))
+    if len(time) > 1:
+    	Dt = time[1] - time[0]
+    else:
+    	Dt = 1
+    for m in range(len(mesh)):
```

Comment:
This could be more readable if your loop was:
```
for m, mat_name, mesh_nodes in enumerate(mesh):
   k[m] = material_property_library[mat_name]['k']
   ....
```

Reply:
This doesn't work as you have it written based on the data structure for mesh, I made a compromised change in the update.